control-plane-operator/hostedclusterconfigoperator/operator/config.go

```diff
@@ -114,8 +115,17 @@ func Mgr(cfg, cpConfig *rest.Config, namespace string) ctrl.Manager {
 				&configv1.ClusterVersion{}:  allSelector,
 				&configv1.FeatureGate{}:     allSelector,
 				&configv1.ClusterOperator{}: allSelector,
+
 				// Needed for inplace upgrader.
 				&corev1.Node{}: allSelector,
+
+				// Needed for resource cleanup
+				&corev1.Service{}:               allSelector,
+				&corev1.Pod{}:                   allSelector,
```

Comment:
Uhm that really sucks, especially the pods? Can we maybe use an API reading client just for the new deletion path?

Reply:
Maybe just for the pods? I do want to watch Services because otherwise I wouldn't necessarily get requeued when they go away.